### DISCUSSION TITLE
katasniplib - library of snippets for Codewars authors and translators

### DISCUSSION THREAD

**Discussion by hobovsky** [codewars/codewars.com#2963] [Show and tell]
For a long time I've been thinking how to make lives of authors, translators, and content maintainers easier, and how to improve quality of kata and their code. There are [Codewars docs](https://docs.codewars.com/authoring) which collect guidelines, authoring guides, some recipes, but still I think they are not reachable easily enough: articles cannot be too bloated to be easily followed, and they still miss out many ideas. Short recipes would be more useful, but navigating back and forth between kata editor and documentation page is not very convenient. That's why I [proposed](https://github.com/codewars/codewars.com/issues/2954#issuecomment-1457138211) at some point a library reachable straight from kata editor, and now made it into a more or less working solution.

The script can be found in [hobovsky/katasniplib](https://github.com/hobovsky/katasniplib) GitHub repository.

The script is meant to provide examples, snippets, and code recipes to users working on a code of a kata or a translation. Snippets are meant to be easily reachable because the library is embedded into Codewars UI and can be used after clicking the `{...}` button in the top right of Codewars code editors:

![image](https://user-images.githubusercontent.com/23709795/224725733-21a86e56-0642-4f88-81fa-6d27162720fb.png)

Currently there is not too many snippets available, but I will keep adding them. Users can contribute to the library by submitting their ideas of snippets to the GitHub repository of the script.

Have fun!

**Comment by Blind4Basics** [comment]
missing technical information/link to install the script?

**Reply by hobovsky** [reply]
🤦 

Edited the post, added a link to the script repo. Thanks.
